skopt/searchcv.py

```diff
@@ -463,14 +510,33 @@ def fit(self, X, y=None, *, groups=None, callback=None, **fit_params):
         else:
             self.optimizer_kwargs_ = dict(self.optimizer_kwargs)
 
+        if callable(self.refit):
+            raise ValueError("BayesSearchCV doesn't support callable refit")
```

Comment:
After having a deeper look into `BayesSearchCV` and skopt, I don't think this is the case. Here is my understanding:

The `tell()` method of the optimizer is used to feed back the values of the objective function at the proposed points. In this context, the points are in the `params` local variable. We also need the value of the objective function at those points, which are retrieved in the `local_results`.

Now, `refit` job isn't actually to provide an objective function, but to choose the best-performing (hyper)parameters among a set, identified in-code by the `best_index_` of `BaseSearchCV`. When `refit` is a str, this is done by an argmax on a score, so refit implicitly uses the following steps : `params -> score() -> argmax() -> best_index`. In that case, multi-metric `BayesSearchCV` can piggy-back on that definition to retrieve the score, and optimize on that. Since the next step after the score is an argmax, maximizing the score improves the best param in a quantifiable way.

If `refit` is a callable, there no longer is a score step, the computation is simply `params -> refit() -> best_index`. This no longer defines a score we can use as an objective function, and in that case the semantic of `refit` and the objective function no longer overlap. This is reflected in `BayesSearchCV` in that `best_score_` cannot be defined when using a callable `refit`.

From a pure function signature point of view, `optimizer.tell()` expect as `y` argument an iterable the same length as `params`, but `all_results['mean_test_score'][best_index]` is of length 1. Another way to see it is saying that `all_result` first index is on the score, and the second is on time. So the `[-len(params):]` is the only way to get "the latest N scores". When defining `local_results`, the question is "which score", so only the first indexing should be modified.

let me know if this makes sense.